sdks/java/core/src/main/java/org/apache/beam/sdk/io/Source.java

```diff
@@ -64,6 +66,7 @@
   /**
    * Returns the default {@code Coder} to use for the data read from this source.
    */
+  @Nullable
```

Comment:
Now that I think of it, it would be better to have consistency between PTransform.getDefaultOutputCoder (throws CannotProvideCoderException if it can't provide a coder) and Source.getDefaultOutputCoder - the latter should probably also throw a CannotProvideCoderException, rather than be nullable.